librdkafka dependency and setup issues

**PyKafka version**: 2.8.0

In setup.py, with CPython, it detects the existence of librdkafka and compile the extension by default. When it fails, it silently disables the librdkafka support.

When I set up the environment for librdkafka, I installed the system `librdkafka-dev` then I ran setup.py. And I actually bump into a silent error when I ran `pip install -e .`. I had to run `setup.py install` to see the error message:
```
running build_ext
building 'pykafka.rdkafka._rd_kafka' extension
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fno-strict-aliasing -Wdate-time -D_FORTIFY_SOURCE=2 -g -fstack-protector-strong -Wformat -Werror=format-security -fPIC -I/usr/include/python2.7 -c pykafka/rdkafka/_rd_kafkamodule.c -o build/temp.linux-x86_64-2.7/pykafka/rdkafka/_rd_kafkamodule.o
pykafka/rdkafka/_rd_kafkamodule.c: In function 'Producer_delivery_report_callback':
pykafka/rdkafka/_rd_kafkamodule.c:568:9: error: unknown type name 'rd_kafka_timestamp_type_t'
        rd_kafka_timestamp_type_t timestamp_type;
        ^
pykafka/rdkafka/_rd_kafkamodule.c:569:29: warning: implicit declaration of function 'rd_kafka_message_timestamp' [-Wimplicit-function-declaration]
        int64_t timestamp = rd_kafka_message_timestamp(rkmessage, &timestamp_ty
                            ^
pykafka/rdkafka/_rd_kafkamodule.c:570:31: error: 'RD_KAFKA_TIMESTAMP_NOT_AVAILABLE' undeclared (first use in this function)
        if (timestamp_type != RD_KAFKA_TIMESTAMP_NOT_AVAILABLE) {
                              ^
pykafka/rdkafka/_rd_kafkamodule.c:570:31: note: each undeclared identifier is reported only once for each function it appears in
pykafka/rdkafka/_rd_kafkamodule.c: In function 'Consumer_consume':
pykafka/rdkafka/_rd_kafkamodule.c:978:9: error: unknown type name 'rd_kafka_timestamp_type_t'
        rd_kafka_timestamp_type_t timestamp_type;
        ^
pykafka/rdkafka/_rd_kafkamodule.c:980:31: error: 'RD_KAFKA_TIMESTAMP_NOT_AVAILABLE' undeclared (first use in this function)
        if (timestamp_type == RD_KAFKA_TIMESTAMP_NOT_AVAILABLE) {
```

1. Is it possible to let users specify rdkafka support and fail the setup if the compilation fails? If the user expects rdkafa support, failing at setup sounds better than failing with ImportError at runtime.
2. Should we document the version of librdkafka needed for the extension to compile successfully? I got librdkafka 0.8.6 (the latest in apt for Ubuntu 16.04) but it doesn't compile. When I manually compile the v0.11.6 from GitHub, it works. There's a similar issue in confluent-kafka-python https://github.com/confluentinc/confluent-kafka-python/issues/230
3. Should we vendor librdkafka? git clone, git checkout to tag, compile and use that shared library file (or use extension)?

I could help with the fix and the doc but there are some decisions to be made. Thanks!

--- a/README.rst
+++ b/README.rst
@@ -159,8 +159,16 @@ Using the librdkafka extension
 ------------------------------
 
 PyKafka includes a C extension that makes use of librdkafka to speed up producer
-and consumer operation. PyKafka requires librdkafka v0.9.1+. Some system package managers may 
-not have up-to-date versions. To use the librdkafka extension, you need to make sure the header
+and consumer operation.
+
+To ensure the C extension is compiled, set environment variable ``RDKAFKA_INSTALL=system`` during
+``pip install`` or ``setup.py``, i.e. ``RDKAFKA_INSTALL=system pip install pykafka``.
+The setup will fail if C extension is not compiled. Oppositely, if ``RDKAFKA_INSTALL=''``, this
+explicitly specifies that the C extension should not be compiled. The current default behavior is
+to compile the extension but will not fail the setup if compilation fails.
+
+PyKafka requires librdkafka v0.9.1+. Some system package managers may not have
+up-to-date versions. To use the librdkafka extension, you need to make sure the header
 files and shared library are somewhere where python can find them, both when you build
 the extension (which is taken care of by ``setup.py develop``) and at run time.
 Typically, this means that you need to either install librdkafka in a place

--- a/setup.py
+++ b/setup.py
@@ -188,13 +188,13 @@ def run_setup(with_rdkafka=True):
 # Use environment variable RDKAFKA_INSTALL to explicitly specify whether rdkafka c extension should be compiled
 # 'system': Compile with librdkafka installed in system
 # '' (empty string): No compile
-install_type = os.environ.get('RDKAFKA_INSTALL')
+rdkafka_install_type = os.environ.get('RDKAFKA_INSTALL')
 
 try:
-    if install_type == '':
+    if rdkafka_install_type == '':
         run_setup(with_rdkafka=False)
     elif not is_cpython:
-        if install_type == 'system':
+        if rdkafka_install_type == 'system':
             raise Exception("librdkafka is not supported under %s, but RDKAFKA_INSTALL specified"
                             % python_implementation)
         print("librdkafka is not supported under %s" % python_implementation)
@@ -204,9 +204,9 @@ try:
         print(15 * "-")
         run_setup(with_rdkafka=False)
     else:
-        run_setup()  # Try to compile c ext for cpython by default
+        run_setup()
 except ve_build_ext.BuildFailed as exc:
-    if install_type == 'system':
+    if rdkafka_install_type == 'system':
         raise
     print(15 * "-")
     print("INFO: Failed to build rdkafka extension:")

--- a/setup.py
+++ b/setup.py
@@ -184,8 +184,19 @@ def run_setup(with_rdkafka=True):
         ]
     )
 
+
+# Use environment variable RDKAFKA_INSTALL to explicitly specify whether rdkafka c extension should be compiled
+# 'system': Compile with librdkafka installed in system
+# '' (empty string): No compile
+install_type = os.environ.get('RDKAFKA_INSTALL')
+
 try:
-    if not is_cpython:
+    if install_type == '':
+        run_setup(with_rdkafka=False)
+    elif not is_cpython:
+        if install_type == 'system':
+            raise Exception("librdkafka is not supported under %s, but RDKAFKA_INSTALL specified"
+                            % python_implementation)
         print("librdkafka is not supported under %s" % python_implementation)
         print(15 * "-")
         print("INFO: Failed to build rdkafka extension:")
@@ -193,8 +204,10 @@ try:
         print(15 * "-")
         run_setup(with_rdkafka=False)
     else:
-        run_setup()
+        run_setup()  # Try to compile c ext for cpython by default
 except ve_build_ext.BuildFailed as exc:
+    if install_type == 'system':
+        raise
     print(15 * "-")
     print("INFO: Failed to build rdkafka extension:")
     print(exc.cause)

